src/zenml/zen_stores/migrations/versions/389046140cad_data_versioning.py

```diff
@@ -83,20 +83,48 @@ def upgrade() -> None:
         )
 
     # Model Version Artifacts and Runs Tables
+    # Drop all entries with null IDs
+    op.execute(
```

Comment:
In the cloud case where we discovered this issue, there wasn't actually any entry where the `artifact_id` or `pipeline_run_id` is `NULL`, but the issue was related to the lower fix here which deletes/recreates the foreign key. Do we still want this to happen, as it potentially deletes user data that might be relevant to them?

In my pipeline run migration, I encountered a similar case and instead
- checked whether there were any entries that would fail the constraint
- raise an exception which explains the reason why the migration failed

If we do this as the initial step of the migration, nothing else will happen and users can just re-run it once they fixed the erroneous entries.

Reply:
It's just links of a model version to an artifact and any links without such IDs would be completely useless and even break the dashboard, so I think we should drop these regardless

Reply:
How it happen that `artifact_id` is null in this table? It is not possible according to the Request models. This might be a sign of a bigger issue, rather than just something we need to fix with a plug like that. Can I have a look at some instances in such a state?

Reply:
It should cascade then
```python
    model_versions_artifacts_links: List[
        "ModelVersionArtifactSchema"
    ] = Relationship(
        back_populates="artifact",
        sa_relationship_kwargs={"cascade": "delete"},
    )
```